packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -0,0 +1,50 @@
+interface AnalyticsEvent {
+    name: string;
+    properties: {
+        [key: string]: any;
+    };
+}
+
+let eventQueue: AnalyticsEvent[] = [];
+
+const setEvent = (event: AnalyticsEvent): void => {
+    eventQueue.push(event);
+    localStorage.setItem('pending_events', JSON.stringify(eventQueue));
+};
+
+const trackEvent = (event: AnalyticsEvent): void => {
+    if (window.rudderanalytics) {
+        window.rudderanalytics.track(event.name, event.properties);
+    } else {
+        setEvent(event);
+    }
+};
+
+const loadPendingEvents = (): void => {
+    const storedEvents = localStorage.getItem('pending_events');
+
+    if (storedEvents) {
+        eventQueue = JSON.parse(storedEvents) as AnalyticsEvent[];
+    }
+};
+
+const initializeAnalytics = async (): Promise<void> => {
```

Comment:
```suggestion
const handleCachedEvents = async (): Promise<void> => {
```
also this must be called in the trackEventWithCache function.
so you dont need to do it manually.
You can also export so called `handleCachedEvents` as well so in case its needed we call it separately.